### PR TITLE
Add meta to WordPress plugin landing page

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR adds the SEO title and og:social card tags.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203363157432084/1203553091426367/f) _(internal)_
- [Asana task](https://app.asana.com/0/1203363157432084/1203553091426361/f) _(internal)_

### CODE REVIEW

File: apps/site/src/pages/wordpress.page.tsx
Comment (CiaranMn) [inline apps/site/src/pages/wordpress.page.tsx:22]: You don't need to specify these, they'll be taken from the defaults
```suggestion
```

**Review by CiaranMn — APPROVED** [review]

File: apps/site/src/pages/wordpress.page.tsx
Comment (nonparibus) [inline apps/site/src/pages/wordpress.page.tsx:15]:
```suggestion
        title: "Block Protocol plugin for WordPress",
```

This looked worse than I thought in most preview environments 😆  Sorry

**Comment by nonparibus** [comment]
@luisbettencourt Can you resolve conflicts? This is good to merge pending approval then. Functional LGTM

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**
